text/0000-multiarch-builders-and-package.md

```diff
@@ -0,0 +1,944 @@
+As we can see, the proposal is based on the assumption that the `run-image`, `build-image` and `buildpacks` to include 
+in the builder are **multi-arch artifacts**, and we can reach them by reading an 
+[image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md)
+
+```bash
+pack builder create my-jammy-builder --config ./builder.toml \ 
+      --multi-arch \ 
+      --publish
+```
+
+In this case, because `targets` are specified and `--multi-arch` is being used `pack` will follow the builder creation process for **each provided platform**, 
+pulling the correct (based on os/arch) buildpacks, build and run images and creating different builders images that are 
+exported and combined into an [image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md)
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Buildpack Package
+
+As a quick summary, our current process to create a buildpack package involves:
+
+- The end-users defined `os` for the OCI image using the [package.toml](https://buildpacks.io/docs/reference/config/package-config/) when they want to export a buildpack package.
+- The only values allowed are `linux` and `windows` and by default when is not present, `linux` is being used.
+- When exporting to daemon, the `docker.OSType` must be equal to `platform.os`
+
+### To keep compatibility 
+
+We propose:
+- Deprecate the `platform.os` field from [package.toml](https://buildpacks.io/docs/reference/config/package-config/). It will be removed after two pack releases with the new feature
+- When `platform.os` is present in [package.toml](https://buildpacks.io/docs/reference/config/package-config/), throw a warning messages indicating the field will be removed 
+and `--platform` flag must be used
+- When `platform.os` is not present [package.toml](https://buildpacks.io/docs/reference/config/package-config/) and `--platform` flag is not used, throw a warning messages indicating 
+a new `--platform` flag is available
+- Keep doing our current process to package a buildpack
+
+### To improve user experience
+
+We propose:
+- Add a new `--platform` flag with format `[os][/arch][/variant]:[name@version]` to build for a particular target, once the `platform.os` field is removed, 
+this will be the way for end-users to specify the platform for which they want to create single OCI artifact.
+> **Note**
+> We want to avoid duplicated information, we expect to have `targets` with all the platform supported by a buildpack
+
+- Add a new boolean `--multi-arch` flag to indicate pack it must create multiples OCI artifacts and combine them with an [image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md), 
+this flag must be used in conjunction with `--publish` or `--format file` flags and error out when `daemon` is selected.
+> **Note**
+> I got feedback to avoid creating an Image Index if users are not expecting that, the flag is an acknowledgment for creating the index 
```

Comment:
Similar thoughts here. Could this be inferred based on the presence of multiple targets? I get that we don't want to just suddenly start making image indexes as that changes things. But couldn't we only do this if multiple targets with different arch's are specified? Or is there a case where you'd have multiple target/arch's and still not want that behavior?

Reply:
I am trying to think of who would be specifying multiple targets today and I'm guessing it would just be bash buildpacks. Maybe we could print a warning if multiple targets are defined but there is only one ./bin/detect or ./bin/build.

Reply:
The scenarios I was thinking were these:
1. I do not have `targets` on my `buildpack.toml` 
   - In this case, the idea is to `pack buildpack package` do the same thing as it is doing today. 
1. I do have `targets` on my `buildpack.toml` 
  - **Daemon** case: `pack` should build my **buildpack package** with the platform defined by the daemon (as it is doing today) without requiring any new flag. 
  - **Registry or File** case: 
     -  If more than 1 target is defined: `pack` will create the `Image Index`, If I just want to create 1 single package, I will need to use `--target` flag to defined which one.
     - If only 1 target is available, nothing changes